### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by mleisa**
In reference to: https://github.com/giantswarm/giantswarm/issues/23864

This PR adds to the readme:
- contributing guidelines
- release process
- a small description of what cert-manager is
- installation instructions
- index

#### Optional app

- [ ] fresh install
- [ ] upgrade from previous version

#### Pre-installed app

- [ ] fresh install during cluster creation
- [ ] upgrade from previous version in a pre-existing cluster

#### Other testing

- [ ] check reconciliation of existing resources after upgrading

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by AverageMarcus — APPROVED** [review]
Love it! ✅

File: README.md [hunk 1]
```diff
@@ -30,10 +39,25 @@ Configuration options are documented in [Configuration.md](https://github.com/gi
 
 Before upgrading, please check [upgrading.md](https://github.com/giantswarm/cert-manager-app/blob/master/docs/upgrading.md).
 
+## Release Process
+
+* Ensure CHANGELOG.md is up to date.
+* Create a new branch with name `master#release#vX.X.X`. Automation will create a release PR.
```
Comment (Gacko) [inline README.md:45]: release#major/minor/patch is also fine, no specific version needed. Don't mind me if this is copied from a template. :)

File: README.md [hunk 2]
```diff
@@ -30,10 +39,25 @@ Configuration options are documented in [Configuration.md](https://github.com/gi
...
+* Merging the release PR will push a new git tag and trigger a new tarball to be pushed to the
+  [giantswarm-catalog].
+* Test and verify the cert-manager release across supported environments in a new or existing WIP platform release.
+
+## Contributing & Reporting Bugs
+If you have suggestions for how `cloudflared` could be improved, or want to report a bug, open an issue! We'd love all and any contributions.
```
Comment (Gacko) [inline README.md:51]: This should be `cert-manager`, not `cloudflared`, right?

**Review by Gacko — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mleisa) [reply README.md:51]: missed that one, thanks!

File: README.md (hunk 1 above)
Comment (mleisa) [reply README.md:45]: Was a bit torn on which to go with. I think ultimately minor/major/patch is a better option, I'll change it

File: README.md [hunk 3]
```diff
@@ -30,10 +39,25 @@ Configuration options are documented in [Configuration.md](https://github.com/gi
...
+If you have suggestions for how `cert-manager` could be improved, or want to report a bug, open an issue! We'd love all and any contributions.
+
+Check out the [Contributing Guide](https://github.com/giantswarm/cert-manager-app/blob/main/CONTRIBUTING.md) for details on the contribution workflow, submitting patches, and reporting bugs.
+
+
 ---
 
 [app-operator]: https://github.com/giantswarm/app-operator
 [cluster-operator]: https://github.com/giantswarm/cluster-operator
+[cert-manager]: https://github.com/jetstack/cert-manager
```
Comment (ubergesundheit) [inline README.md:60]: Repo was moved to https://github.com/cert-manager/cert-manager 
GitHub redirects, but I think its worth having an up to date URL here

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+# Email and chat
+
+- Email: [giantswarm](https://groups.google.com/forum/#!forum/giantswarm)
+- IRC: #[giantswarm](irc://irc.freenode.org:6667/#giantswarm) IRC channel on freenode.org
+
+## Getting started
+
+- Fork the repository on GitHub
+- Read the [README.md](https://github.com/giantswarm/cert-manager-app/blob/master/README.md) for build instructions
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:13]: We usually don't have building instructions in App readme files. IMO we should include some basic instructions

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# How to contribute
+
+cert-manager is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, contact points for developers and other resources to make getting your contribution into cert-manager easier.
+
+# Email and chat
+
+- Email: [giantswarm](https://groups.google.com/forum/#!forum/giantswarm)
+- IRC: #[giantswarm](irc://irc.freenode.org:6667/#giantswarm) IRC channel on freenode.org
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:8]: Where did you copy this file from?

I guess nobody is lurking in the freenode channel anymore and the google group also looks dead.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+This is a rough outline of what a contributor's workflow looks like:
+
+- Create a feature branch from where you want to base your work. This is usually main.
+- Make commits of logical units.
+- Make sure your commit messages are in the proper format (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- Submit a pull request to giantswarm/cert-manager-app.
+- Adding unit tests will greatly improve the chance for getting a quick review and your PR accepted.
+- Your PR must receive a LGTM from a maintainer.
+- Before merging your PR, be sure to squash all commits into one.
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:46]: We use GitHub for this. IMO having all commits is a great way to see how a PR has progressed

File: CONTRIBUTING.md (hunk 5 above)
Comment (mleisa) [reply CONTRIBUTING.md:8]: A while back I checked a number of repos for contributing guidelines, one of them was [openstack-app-collection](https://github.com/giantswarm/openstack-app-collection/blob/master/CONTRIBUTING.md). I don't remember all others but there were a few.

Seems like this part is outdated and nobody updated it. Any idea where I could find infos about which contact info to replace that with?

Comment (ubergesundheit) [reply CONTRIBUTING.md:8]: The file is probably outdated in most of the repositories (wild guess).

You could think about automatically keeping this file updates through giantswarm/github. Content-wise I have no idea. GitHub is probably the easiest for us, but there are maybe people that don't want to create an account.

File: README.md [hunk 7]
```diff
@@ -42,13 +42,13 @@ Before upgrading, please check [upgrading.md](https://github.com/giantswarm/cert
 ## Release Process
 
 * Ensure CHANGELOG.md is up to date.
-* Create a new branch with name `master#release#vX.X.X`. Automation will create a release PR.
+* Create a new branch to trigger the release workflow as either a patch, minor, or major. E.g. to release a patch, create a branch from master called master#release#patch and push it. Automation will create a release PR.
```
Comment (Gacko) [inline README.md:45]: Just `release#patch` is fine, too.

File: CONTRIBUTING.md (hunk 5 above)
Comment (Gacko) [reply CONTRIBUTING.md:8]: Well... I asked myself the same: Who's using IRC? :D Thought we might have some kind of Slack integration...

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+You might also want to read [Elika Etemad’s article on filing good bug reports](http://fantasai.inkedblade.net/style/talks/filing-good-bugs/) before creating a bug report.
+
+We might ask you for further information to locate a bug. A duplicated bug report will be closed.
+
+## Contribution flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Create a feature branch from where you want to base your work. This is usually main.
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:39]:
```suggestion
- Create a feature branch from where you want to base your work. This is usually master.
```

Comment (Gacko) [reply CONTRIBUTING.md:39]: ... could we rename `master` to `main` in this repository instead? :>

Comment (mleisa) [reply CONTRIBUTING.md:39]: We could, but I don't believe it belongs into the scope of this PR :)

File: README.md [hunk 9]
```diff
@@ -4,7 +4,62 @@
...
+Here is an example that would install the cert-manager-app to workload cluster `abc12`:
+
+```
+# appCR.yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+metadata:
+
+  # workload cluster resources live in a namespace with the same ID as the workload cluster.
+  name: cert-manager-app
```
Comment (webwurst) [inline README.md:39]: We can leave out `-app` since kind is already an `App`
```suggestion
  name: cert-manager
```

File: README.md
Comment (webwurst) [inline README.md:1]: Maybe link out to our app documentation somewhere for more details? https://docs.giantswarm.io/getting-started/app-platform/deploy-app/#creating-an-app-cr

Comment (webwurst) [inline README.md:1]: And maybe also link out to cert-manager upstream? For example to [Issuing Certificates](https://cert-manager.io/docs/usage/)

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -43,7 +43,7 @@ This is a rough outline of what a contributor's workflow looks like:
 - Submit a pull request to giantswarm/cert-manager-app.
 - Adding unit tests will greatly improve the chance for getting a quick review and your PR accepted.
 - Your PR must receive a LGTM from a maintainer.
-- Before merging your PR, be sure to squash all commits into one.
+- Merge your PR
```
Comment (ubergesundheit) [inline CONTRIBUTING.md:46]:
```suggestion
- We merge your PR
```

**Pull request merged**